platform/engineering/design-docs/2020-05-26-content-validation.md

```diff
@@ -0,0 +1,278 @@
+
+### Testing Plan
+
+#### Local
+
+##### Broken links
+
+###### Positive
+
+1. Confirm that the content build has zero broken links
+2. Run the content validation script
+3. Confirm that the script reports zero broken links
+
+###### Negative
+
+1. Confirm that the content build has zero broken links
+2. Modify an HTML file, like [`header.html`](https://github.com/department-of-veterans-affairs/content-build/blob/master/src/site/includes/header.html), to have a broken link
+3. Run the content validation script
+4. Confirm that the script reports the newly introduced broken link
+
+##### Accessibility errors
+
+###### Positive
+
+1. Confirm that the content build has zero accessibility errors
+2. Run the content validation script
+3. Confirm that the script reports zero accessibility errors
+
+###### Negative
+
+1. Confirm that the content build has zero accessibility errors
+2. Modify an HTML file, like [`header.html`](https://github.com/department-of-veterans-affairs/content-build/blob/master/src/site/includes/header.html), to have an accessibility error
+3. Confirm that the script reports the newly introduced accessibility error
+
+#### CI
+
+##### Positive
+
+1. Confirm that the content build has zero accessibility errors and zero broken links
+2. Run the job that calls the content validation script
+3. Confirm that a report is sent to the #cms-team Slack channel
+4. Confirm that the report includes zero accessibility errors and zero broken links
+
+##### Negative
+
+1. Confirm that the content build has zero accessibility errors and zero broken links
+2. Add a broken link, and an accessibility error
+3. Run the job that calls the content validation script
+4. Confirm that a report is sent to the #cms-team Slack channel
+5. Confirm that the report includes the newly introduced accessibility error
+6. Confirm that the report includes the newly introduced broken link
+7. Fix the newly introduced accessibility error
+8. Fix the newly introduced broken link
+9. Run the job that calls the content validation script
+10. Confirm that the report includes zero accessibility errors and zero broken links
+
+### Logging
+
+Currently, invalid content is logged in the build log during staging deploys. And failed builds trigger generic [Slack notifications](https://dsva.slack.com/archives/C37M86Y8G/p1591099447374800).
+
+The updated content validation approach will continue logging invalid content errors in the build log, and will also post an automated report of the invalid content in the #cms-team Slack channel.
+
+### Debugging
+
+- Local build logs
+- Jenkins build logs
+- Slack notifications
+
+### Caveats
+
+- The current broken link checking in Drupal can handle redirects, but those redirects are not currently synchronized with `vets-website`. When the redirects are not synchronized between `vets-website` and Drupal, a page that is moved in Drupal may pass the Drupal broken link check but fail the `vets-website` broken link check.
```

Comment:
what does it mean to say "can handle redirects, but those redirects are not currently synchronized with `vets-website`"?

* does "handle redirects" mean "follow all 301/302s on the live site"? (if so, I don't understand the "synchronized with vets-website" comment)
* does it mean "understands redirects configured within Drupal"? (if so, could you add that?)

Reply:
It means that the redirects from drupal are not being imported into nginx during the content build. 

See the following quote from Steve W. for more info. 
> Where it still has the potential to fall down right now is that if a path to a page in drupal is changed, drupal automatically creates a redirect from the old path to the new path, so when the links are checked, they still work. The original plan was that the Front end would import redirects from drupal into Nginx when the content build takes place. However that is not happening, so the front end does not have the redirects, so a moved page, results in broken links from pre-existing content links. - https://github.com/department-of-veterans-affairs/va.gov-team/pull/9658#discussion_r434864015

I'll update this section to be more clear.